test/source/handler.test.js

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* eslint-env mocha */
+/* eslint-disable no-param-reassign */
+import assert from 'assert';
+import sourceHandler from '../../src/source/handler.js';
+
+describe('Source Handler Tests', () => {
+  it('handles GET requests', async () => {
+    const mockBus = {
+      get: async (path, meta) => {
+        assert.equal(path, 'test/site/hello.html');
+        meta.ContentType = 'text/html';
+        meta.timestamp = '946684800000';
+        meta.ETag = '"some-etag-327"';
+        meta.id = 'dc0b68d8-b3ac-4c8a-9205-d78085e55704';
+        return '<body>Hello, world!</body>';
+      },
+    };
+    const mockStorage = { sourceBus: () => mockBus };
+    const context = { attributes: { storage: mockStorage } };
+    const info = {
+      method: 'GET',
+      org: 'test',
+      site: 'site',
+      resourcePath: '/hello.html',
+    };
+    const result = await sourceHandler(context, info);
+    assert.equal(result.status, 200);
+    assert.equal(await result.text(), '<body>Hello, world!</body>');
+    assert.equal(result.headers.get('Content-Type'), 'text/html');
+    assert.equal(result.headers.get('Content-Length'), '26');
+    assert.equal(result.headers.get('Last-Modified'), 'Sat, 01 Jan 2000 00:00:00 GMT');
+    assert.equal(result.headers.get('ETag'), '"some-etag-327"');
+    assert.equal(result.headers.get('X-da-id'), 'dc0b68d8-b3ac-4c8a-9205-d78085e55704');
+    assert.equal(result.headers.get('Access-Control-Allow-Origin'), '*');
+    assert.equal(result.headers.get('Access-Control-Allow-Headers'), '*');
+    assert.equal(result.headers.get('Access-Control-Allow-Methods'), 'HEAD, GET, PUT, DELETE');
+    assert.equal(result.headers.get('Access-Control-Expose-Headers'), 'X-da-id');
```

Comment:
we prefer comparing headers with:
```
assert.deepStrictEqual(result.headers.plain(), {
  ... all headers
});
```

see: https://github.com/adobe/helix-api-service/blob/030047561fc9bfbc18c510c1f19a14841438b463/test/live/handler.test.js#L110-L114